README.md

```diff
@@ -355,6 +356,57 @@ writable.
 
 Always prefer `ReadonlyArray` over a regular `Array` unless it must be possible to modify the Array.
 
+### React
+
+#### Avoid using `React.FC` and `FunctionComponent` built-ins of `@types/react`
+
+They provide an implicit definition of `children`, which can be misleading to components which do
+not accept `children`. It is encouraged to instead either define `children?: ReactNode` on your prop
+type or wrap your prop type in the `PropsWithChildren` type util.
+
+```ts
+type MonkComponentProps = {
+  // other properties
+  children?: ReactNode;
+};
+
+function MonkComponent({ children }: MonkComponentProps): ReactElement {
+  //...
+}
+
+// with `PropsWithChildren`
+
+type MonkComponentProps = {
+  // other properties
+};
+
+function MonkComponent({ children }: PropsWithChildren<MonkComponentProps>): ReactElement {
```

Comment:
I don't think we should add this example. The example above is a more concise way to set a type for the children prop. This also creates an anonymous type that can only be inferred using eg the `React.ComponentProps<T>` type helper.

Reply:
I don't really see a problem with this, and would actually say that that is an advantage rather than an issue.

1. Each component should decide by itself if it accepts children or not.
2. If I reuse a prop type that has `children?: ReactNode` defined, then I might accidentally permit `children` where I don't want them.

```tsx
type ChildProps = { children?: ReactNode };
function Child({ children }: ChildProps { /* do stuff */ }

type ParentProps = { /* stuff */ } & ChildProps
function Parent(props: ParentProps { /* do stuff */ }

<Parent>/* oops, I can pass children and TS is ok */</ Parent>
```